shared/actions/wallets.js

```diff
@@ -41,12 +44,64 @@ const loadPaymentsSuccess = (res: RPCTypes.PaymentsPageLocal, action: WalletsGen
   )
 }
 
+const linkExistingAccount = (state: TypedState, action: WalletsGen.LinkExistingAccountPayload) => {
+  const {name, secretKey} = action.payload
+  return RPCTypes.localLinkNewWalletAccountLocalRpcPromise(
+    {
+      name,
+      secretKey: secretKey.stringValue(),
+    },
+    Constants.linkExistingWaitingKey
+  )
+    .then(accountID => [
+      WalletsGen.createLoadAccounts(),
+      WalletsGen.createSelectAccount({accountID: Types.stringToAccountID(accountID), show: true}),
+      WalletsGen.createLinkExistingAccountError({error: '', name, secretKey}),
+    ])
+    .catch(err => WalletsGen.createLinkExistingAccountError({error: err.desc, name, secretKey}))
+}
+
+const validateAccountName = (state: TypedState, action: WalletsGen.ValidateAccountNamePayload) => {
+  const {name} = action.payload
+  return RPCTypes.localValidateAccountNameLocalRpcPromise({name})
+    .then(() => WalletsGen.createValidateAccountNameError({error: '', name}))
```

Comment:
i think this should be a different action. this handles the validateAccountName action, and happens to not be called on error but really i think there's 2 semantic actions
1. validateAccountName (aka i want it validated)
2. validatedAccountName (aka i'm done validating)